libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/ui/SalesforceReactActivity.java

```diff
@@ -276,6 +284,25 @@ protected boolean shouldReactBeRunning(){
 
     protected void restartReactNativeApp(){
         SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().destroy();
-        SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().createReactContextInBackground();
+        if(shouldReactBeRunning()){
+            SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().createReactContextInBackground();
+        }
+    }
+
+    private boolean shouldAskOverlayPermission(){
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if(SalesforceReactActivity.this.getReactNativeHost().getReactInstanceManager().getDevSupportManager().getDevSupportEnabled()){
+                if (!Settings.canDrawOverlays(this)) {
```

Comment:
So it happens when the delegate is created. So what's the user experience like, they see the popup and then when they go back, the react activity is resumed, shouldReactBeRunning is true, and react loads?
What if they answered no. Are you stuck with a grey screen?
When shouldAskOverlayPermission returns false, should we ask for permission?